packages/mui-joy/src/IconButton/IconButton.tsx

```diff
@@ -77,7 +77,11 @@ export const StyledIconButton = styled('button')<{ ownerState: IconButtonOwnerSt
       [theme.focus.selector]: theme.focus.default,
     },
     theme.variants[ownerState.variant!]?.[ownerState.color!],
-    { '&:hover': theme.variants[`${ownerState.variant!}Hover`]?.[ownerState.color!] },
+    {
+      '&:hover': {
+        '@media (hover: hover)': theme.variants[`${ownerState.variant!}Hover`]?.[ownerState.color!],
```

Comment:
In Material UI, I don't recall why we used the opposite: https://github.com/mui/material-ui/issues/15000#issuecomment-689684361. It might have been because of IE 11 support https://caniuse.com/mdn-css_at-rules_media_hover. I think that it could be good to normalize on what works best.